recipes/mpdecimal/2.5.x/conanfile.py

```diff
@@ -219,7 +215,10 @@ def package(self):
     def package_info(self):
         lib_pre_suf = ("", "")
         if is_msvc(self):
-            lib_pre_suf = ("lib", f"-{self.version}")
+            if self.options.shared:
+                lib_pre_suf = ("lib", f"-{self.version}.dll")
```

Comment:
I can see CI is not building windows + shared - so would need some logs that show that these are the files created `.dll.lib` on Windows n-

otherwise, we may as well consider - if a shared libmpdec++ is *never* possible on Windows - we could just make it the default on Windows so that it's not set to be built?

Reply:
In #25903 I actually partially remove the restriction, it is in fact possible to mix these options as of 2.5.1. I'll see what I can find for logs.

Reply:
In this log you can see the generated lib names:
https://c3i.jfrog.io/artifactory/cci-build-logs/cci/prod/PR-25889/6/package_build_logs/build_log_mpdecimal_2_5_1_254318b465fb6c8ce4fdfbadfde99a36_056ff9a90cff2a089a33d9c99cd4b6d10281a6b1.success.txt

You can see in the link step:
```
/out:libmpdec-2.5.1.dll /implib:libmpdec-2.5.1.dll.lib
```
and in the package step:
```
mpdecimal/2.5.1: package(): Packaged 1 '.dll' file: libmpdec-2.5.1.dll
mpdecimal/2.5.1: package(): Packaged 1 '.h' file: mpdecimal.h
mpdecimal/2.5.1: package(): Packaged 1 '.lib' file: libmpdec-2.5.1.dll.lib
```

Reply:
awesome!

Reply:
PR LGTM once we restore the default value of cxx :D